File: Code/Languages/C/flip-vertical-axis.c
// 3/6/2020 - Unknown whether this works

#include <stdio.h>

void print_matrix(int matrix[][], int m) {
    for (int i = 0; i < m; i++) {
        printf("%s\n", matrix[i]);
    }
}

void flip_vertical_axis(int matrix[][], int m, int n) {
    printf("\nGiven matrix:\n");
    print_matrix(matrix, m);
}

int main(void) {
    int matrix_2_2[2][2] = [
        [1,0],
        [1,0]
    ]

    int matrix_2_3[2][3] = [
        [1, 2, 3],
        [1, 2, 3]
    ]

    flip_vertical_axis(matrix_2_2, 2, 2)
    flip_vertical_axis(matrix_2_3, 2, 3)

    return 0;
}

File: Code/Languages/Python/FlipVerticalAxis.py
'''
    Kevin Tran
    Flip a m x n 2D image matrix in-place along its vertical axis.
'''

matrix_2_2 = [
    [1,0],
    [1,0]
]

matrix_2_3 = [
    [1, 2, 3],
    [1, 2, 3]
]

def print_matrix(matrix):
    for row in matrix:
        print row
    print '\n'

def flip_vertical_axis(matrix):
    print '\nGiven matrix:'
    print_matrix(matrix)
    
    for i in range(len(matrix)):
        matrix[i] = matrix[i][::-1]
    
    print 'Flipped matrix:'
    print_matrix(matrix)
    print '========================\n'

flip_vertical_axis(matrix_2_2)
flip_vertical_axis(matrix_2_3)
